js_modules/dagit/packages/core/src/ui/Tabs.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import styled from 'styled-components/macro';
+
+import {ColorsWIP} from './Colors';
+import {IconWrapper} from './Icon';
+import {FontFamily} from './styles';
+
+interface TabsProps {
+  children: Array<React.ReactElement<TabProps>>;
+  selectedTabId?: string;
+  onChange?: (selectedTabId: string) => void;
+  small?: boolean;
+}
+
+export const Tabs = styled(({selectedTabId, children, small, onChange, ...rest}) => {
+  if (!(children instanceof Array) || children.some((c) => c.type !== Tab)) {
+    throw new Error('Tabs must render Tab instances');
+  }
+
+  return (
+    <div {...rest} role="tablist">
+      {children.map((child) =>
+        React.cloneElement(child, {
```

Comment:
Safer to use `React.Children.map`?

Reply:
Oh interesting I guess so - I hadn't seen that before. It looks like I still need this cloneElement line though?